webui/react/src/components/FilterForm/components/FilterFormStore.ts

```diff
@@ -0,0 +1,191 @@
+import { observable, Observable, WritableObservable } from 'micro-observables';
+import { v4 as uuidv4 } from 'uuid';
+
+import {
+  AvailableOperators,
+  Conjunction,
+  FilterFormSet,
+  FilterFormSetWithoutId,
+  FormField,
+  FormFieldValue,
+  FormGroup,
+  FormKind,
+  Operator,
+} from 'components/FilterForm/components/type';
+import { V1ColumnType, V1LocationType, V1ProjectColumn } from 'services/api-ts-sdk';
+
+export const ITEM_LIMIT = 50;
+
+const INIT_FORMSET: Readonly<FilterFormSet> = {
+  filterGroup: { children: [], conjunction: Conjunction.And, id: 'ROOT', kind: FormKind.Group },
+  showArchived: false,
+};
+
+const getInitGroup = (): FormGroup => ({
+  children: [],
+  conjunction: Conjunction.And,
+  id: uuidv4(),
+  kind: FormKind.Group,
+});
+
+const getInitField = (): FormField => ({
+  columnName: 'name',
+  id: uuidv4(),
+  kind: FormKind.Field,
+  location: V1LocationType.EXPERIMENT,
+  operator: AvailableOperators[V1ColumnType.TEXT][0],
+  type: V1ColumnType.TEXT,
+  value: null,
+});
+
+export class FilterFormStore {
+  #formset: WritableObservable<FilterFormSet>;
+
+  constructor(data?: Readonly<FilterFormSet>) {
+    this.init(data);
+  }
+
+  public init(data?: Readonly<FilterFormSet>): void {
+    this.#formset = observable<FilterFormSet>(structuredClone(data ? data : INIT_FORMSET));
```

Comment:
we'll need a shim for this for testing because this global isn't in our minimum version of node.

Reply:
i have node update PR #6759. when its merged, i'll rebase commits